ucm2/Meson/p241/p241-HiFi.conf

```diff
@@ -0,0 +1,38 @@
+SectionVerb {
+	Value {
+		PlaybackPriority 100
+		PlaybackPCM "hw:${CardId},0"
```

Comment:
Move this (both) values to the section device configuration. Also, the priorities should be different (probably analog should have higher priority in this case - missing jack / link detection).

Reply:
Well the pcm is not really linked the output device as I was explaining above.

The card behind this is a DPCM one. The PCMs are the frontends.
The output devices such as HDMI and Analog are the backend.
So in theory, the output device could be attached to different PCMs so I'm not sure it is correct to define the PCM there. 

There is a single PCM in this example - but I have another SoC getting ready where I have several generic PCMs and several outputs. The idea is too use 1 PCM per use-case (Hifi, Voice,) and route the PCM to appropriate output depending on the current use-case.

How would you handle this with UCM if verb is not appropriate ? 

---
For the priority, on these boards, the HDMI is used mostly. If I need to provide a different, I would suggest to put that higher.
There no jack detect on analog, indeed.
Ideally, I'd like the default configuration (priority?) to enable both outputs on the PCM. The HW supports it.

Reply:
The PA or PW use the PCM device name as the sink (thus if those names are equal - one PCM sink is created). Your configuration is also ok, but I would prefer to specify the PCM devices in the SectionDevice values.

PA or PW selects the default device by priority (and if equal - by the device order). If you like to enable both - you may create third conflicting device with appropriate control settings (enable both paths) with highest priority, but I would prefer to keep it separate and let PA or PW to handle it depending on the user selection. If the devices are not marked as conflicting, the applications may use both simultaneously.

> For the priority, on these boards, the HDMI is used mostly. If I need to provide a different, I would suggest to put that higher.

As you like.

Reply:
The priority is connected to the PCM device + mixer / path settings. If this combination is in a clash with another UCM device (PCM device + mixer / path settings), it should be marked as conflicting. The dynamic PCM routing in the driver should be described using a model which fits for the users - it usually means the static allocation of PCM devices for a purpose.

Note that only one verb can be active at a time, so it's something like a profile in pulseaudio.